packages/framework/presence/src/presenceStates.ts

```diff
@@ -194,36 +221,60 @@ export function mergeUntrackedDatastore(
 	}
 }
 
+/**
+ * Produces the value type of a schema element or set of elements.
+ */
+type SchemaElementValueType<
+	TSchema extends PresenceStatesSchema,
+	Keys extends keyof TSchema & string,
+> = Exclude<MapSchemaElement<TSchema, "initialData", Keys>, undefined>["value"];
+
 class PresenceStatesImpl<TSchema extends PresenceStatesSchema>
 	implements
 		PresenceStatesInternal,
 		PresenceStates<TSchema>,
 		StateDatastore<
 			keyof TSchema & string,
-			MapSchemaElement<TSchema, "value", keyof TSchema & string>
+			SchemaElementValueType<TSchema, keyof TSchema & string>
 		>
 {
 	private readonly nodes: MapEntries<TSchema>;
 	public readonly props: PresenceStates<TSchema>["props"];
 
+	public readonly controls: RequiredBroadcastControl;
+
 	public constructor(
 		private readonly runtime: PresenceRuntime,
 		private readonly datastore: ValueElementMap<TSchema>,
 		initialContent: TSchema,
+		controlsSettings: BroadcastControlSettings | undefined,
 	) {
+		this.controls = new RequiredBroadcastControl(/* allowableUpdateLatencyMs */ 60);
+		if (controlsSettings?.allowableUpdateLatencyMs !== undefined) {
+			this.controls.allowableUpdateLatencyMs = controlsSettings.allowableUpdateLatencyMs;
+		}
+
 		// Prepare initial map content from initial state
 		{
 			const clientSessionId = this.runtime.clientSessionId;
 			let anyInitialValues = false;
+			let cumulativeAllowableUpdateLatencyMs: number | undefined;
 			// eslint-disable-next-line unicorn/no-array-reduce
 			const initial = Object.entries(initialContent).reduce(
 				(acc, [key, nodeFactory]) => {
 					const newNodeData = nodeFactory(key, handleFromDatastore(this));
 					acc.nodes[key as keyof TSchema] = newNodeData.manager;
-					if ("value" in newNodeData) {
+					if ("initialData" in newNodeData) {
+						const { value, allowableUpdateLatencyMs } = newNodeData.initialData;
 						acc.datastore[key] = acc.datastore[key] ?? {};
-						acc.datastore[key][clientSessionId] = newNodeData.value;
-						acc.newValues[key] = newNodeData.value;
+						acc.datastore[key][clientSessionId] = value;
+						acc.newValues[key] = value;
+						if (allowableUpdateLatencyMs !== undefined) {
+							cumulativeAllowableUpdateLatencyMs =
+								cumulativeAllowableUpdateLatencyMs === undefined
+									? allowableUpdateLatencyMs
+									: Math.min(cumulativeAllowableUpdateLatencyMs, allowableUpdateLatencyMs);
+						}
```

Comment:
Not for this PR, but is there a particular advantage to using reduce here? I find reduce hard to read myself. This might be a Tyler problem, so take with a grain of salt. :) I asked copilot to rewrite it as a for-loop, and it produced this:

```ts
const initial = { nodes: {}, datastore: {}, newValues: {} };
for (const [key, nodeFactory] of Object.entries(initialContent)) {
    const newNodeData = nodeFactory(key, handleFromDatastore(this));
    initial.nodes[key as keyof TSchema] = newNodeData.manager;
    if ("initialData" in newNodeData) {
        const { value, allowableUpdateLatencyMs } = newNodeData.initialData;
        initial.datastore[key] = initial.datastore[key] ?? {};
        initial.datastore[key][clientSessionId] = value;
        initial.newValues[key] = value;
        if (allowableUpdateLatencyMs !== undefined) {
            cumulativeAllowableUpdateLatencyMs = cumulativeAllowableUpdateLatencyMs === undefined
                ? allowableUpdateLatencyMs
                : Math.min(cumulativeAllowableUpdateLatencyMs, allowableUpdateLatencyMs);
        }
        anyInitialValues = true;
    }
}
```

Completely untested, but it was a little easier to read for me at least. From the looks of it though this is likely losing typing, and perhaps reduce helps there?

Reply:
That copilot fix is incorrect (besides losing type info).
`reduce` can lead to better type results, but I don't think it is any more. It may have been more helpful in earlier versions.
I uncommented the linter `reduce` use suppression and let copilot fix that. It produced something that looks correct to me and kept types. I can follow-up after.